### ISSUE TITLE
`do`/`while` issue

### ISSUE THREAD

**Issue by ike709** [wixoaGit/OpenDream#258]
```
#define SET_SELF(SETVAR) do { \
    var/icon/SELF_ICON=icon(icon(curicon, curstate, base_icon_dir),"",SOUTH,no_anim?1:null); \
    if(A.alpha<255) { \
        SELF_ICON.Blend(rgb(255,255,255,A.alpha),ICON_MULTIPLY);\
    } \
    if(A.color) { \
        if(islist(A.color)){ \
            SELF_ICON.MapColors(arglist(A.color))} \
        else{ \
            SELF_ICON.Blend(A.color,ICON_MULTIPLY)} \
    } \
    ##SETVAR=SELF_ICON;\
    } while (0)
#define INDEX_X_LOW 1
#define INDEX_X_HIGH 2
#define INDEX_Y_LOW 3
#define INDEX_Y_HIGH 4
```

![image](https://user-images.githubusercontent.com/5714543/127758526-806faa5d-6a21-4d90-b6a7-aa58652b245b.png)

**Comment by ike709** [comment]
BeeStation

**Comment by ike709** [comment]
Near-identical exception:
```
```
#define ADD_TRAIT(target, trait, source) \
    do { \
        var/list/_L; \
        if (!target.status_traits) { \
            target.status_traits = list(); \
            _L = target.status_traits; \
            _L[trait] = list(source); \
            SEND_SIGNAL(target, SIGNAL_ADDTRAIT(trait), trait); \
        } else { \
            _L = target.status_traits; \
            if (_L[trait]) { \
                _L[trait] |= list(source); \
            } else { \
                _L[trait] = list(source); \
                SEND_SIGNAL(target, SIGNAL_ADDTRAIT(trait), trait); \
            } \
        } \
    } while (0)
```

**Comment by ike709** [comment]
Again;
```
#define REMOVE_TRAIT(target, trait, sources) \
	do { \
		var/list/_L = target.status_traits; \
		var/list/_S; \
		if (sources && !islist(sources)) { \
			_S = list(sources); \
		} else { \
			_S = sources\
		}; \
		if (_L && _L[trait]) { \
			for (var/_T in _L[trait]) { \
				if ((!_S && (_T != ROUNDSTART_TRAIT)) || (_T in _S)) { \
					_L[trait] -= _T \
				} \
			};\
			if (!length(_L[trait])) { \
				_L -= trait; \
				SEND_SIGNAL(target, SIGNAL_REMOVETRAIT(trait), trait); \
			}; \
			if (!length(_L)) { \
				target.status_traits = null \
			}; \
		} \
	} while (0)
	```

**Comment by wixoaGit** [comment]
Fixed by e4890f8e741fe98f26cfae23f221d7d5c38b9ed8.
